src/QueryReflection/PDOQueryReflector.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace staabm\PHPStanDba\QueryReflection;
+
+use PDO;
+use PDOException;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\Constant\ConstantArrayTypeBuilder;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\Type;
+use staabm\PHPStanDba\Error;
+use staabm\PHPStanDba\TypeMapping\TypeMapper;
+
+final class PDOQueryReflector implements QueryReflector
+{
+    private const PSQL_INVALID_TEXT_REPRESENTATION = '22P02';
+    private const PSQL_UNDEFINED_COLUMN = '42703';
+    private const PSQL_UNDEFINED_TABLE = '42P01';
+
+    private const SUPPORTED_PDO_CODES = [
+        self::PSQL_INVALID_TEXT_REPRESENTATION,
+        self::PSQL_UNDEFINED_COLUMN,
+        self::PSQL_UNDEFINED_TABLE,
+    ];
+
+    private const MAX_CACHE_SIZE = 50;
+
+    /**
+     * @var array<string, PDOException|array|false>
+     */
+    private array $cache = [];
+
+    public function __construct(private PDO $pdo, private TypeMapper $typeMapper)
```

Comment:
the mapper should not be a constructor argument, as these reflector classes will be used within the `phpstan-bootstrap-dba.php` config file and the dba enduser should not be required to pass in classes which he should not know about.

just create and assign the object within the constructor